healvis/simulator.py

```diff
@@ -84,56 +93,164 @@ def parse_telescope_params(tele_params):
-    return_dictionary['integration_time'] = np.ones(time_params['Ntimes'], dtype=np.float)
-    return_dictionary['time_cadence'] = time_params['time_cadence']
-    return_dictionary['time_array'] = time_arr
+    return_dict = {}
 
-    return return_dictionary
+    time_keywords = ['start_time', 'end_time', 'Ntimes', 'time_cadence',
+                     'duration_hours', 'duration_days']
+    st, et, nt, it, dh, dd = [tk in time_params for tk in time_keywords]
+    kws_used = ", ".join(time_params.keys())
+    daysperhour = 1 / 24.
+    hourspersec = 1 / 60.**2
+    dayspersec = daysperhour * hourspersec
+
+    if dh and not dd:
+        time_params['duration'] = time_params['duration_hours'] * daysperhour
+        dd = True
+    elif dd:
+        time_params['duration'] = time_params['duration_days']
+
+    if not nt:
+        if not it:
+            raise ValueError("Either time or Ntimes must be "
+                             "included in parameters:" + kws_used)
+        if st and et:
+            time_params['duration'] = time_params['end_time'] - time_params['start_time'] + time_params['time_cadence'] * dayspersec
+            dd = True
+        if dd:
+            time_params['Ntimes'] = int(np.round(time_params['duration']
+                                                 / (time_params['time_cadence'] * dayspersec)))
+        else:
+            raise ValueError("Either duration or time bounds must be specified: "
+                             + kws_used)
+
+    if not it:
+        if not dd:
+            raise ValueError("Either duration or integration time "
+                             "must be specified: " + kws_used)
+        time_params['time_cadence'] = (time_params['duration'] / dayspersec
+                                       / float(time_params['Ntimes']))  # In seconds
+
+    inttime_days = time_params['time_cadence'] * dayspersec
+    if not dd:
+        time_params['duration'] = inttime_days * (time_params['Ntimes'])
+        dd = True
+    if not st:
+        if et and dd:
+            time_params['start_time'] = time_params['end_time'] - time_params['duration'] + inttime_days
+    if not et:
+        if st and dd:
+            time_params['end_time'] = time_params['start_time'] + time_params['duration'] - inttime_days
+    if not (st or et):
+        raise ValueError("Either a start or end time must be specified: " + kws_used)
+
+    time_arr = np.linspace(time_params['start_time'],
+                           time_params['end_time'] + inttime_days,
+                           time_params['Ntimes'], endpoint=False)
+
+    if time_params['Ntimes'] != 1:
+        assert np.allclose(np.diff(time_arr), inttime_days * np.ones(time_params["Ntimes"] - 1), atol=dayspersec)   # To nearest second
+
+    return_dict['time_cadence'] = time_params['time_cadence']
+    return_dict['time_array'] = time_arr
+    return_dict['Ntimes'] = time_params['Ntimes']
+    return_dict['Nspws'] = 1
```

Comment:
I see this is copy/paste from the `pyuvsim` `simsetup` module. I left these lines out on purpose b/c I think they do more harm then good. I don't see a point in returning a notion of `spws` and `pols` when they weren't requested for and aren't needed in the context of how this function is used. I would opt to say we don't return populate these keys.